victoriensularieh.ConsoleHabitTracker/Menu.cs

```diff
@@ -0,0 +1,464 @@
+class Menu
+{
+    public static void ShowGreeting()
+    {
+        Console.WriteLine("Hello there User!\nHow can i be of assisstance?\n");
+        Console.WriteLine("Press any key to access the Main menu.\n");
+        Console.ReadLine();
+        Console.Clear();
+    }
+
+    public static void ShowMainMenu()
+    {
+        string choice = "";
+        bool rightChoice = false;
+
+        Console.WriteLine(@"Press A to view habits Menu.
+Press B to view system units Menu.
+Press C to View entries Menu.
+Press X to Exit.");
+
+        while (rightChoice == false)
+        {
+            choice = Helpers.ValidateStringInput();
+            switch (choice.ToUpper())
+            {
+                case "A":
+                    ShowHabitMenu();
+                    rightChoice = true;
+                    break;
+                case "B":
+                    ShowUnitMenu();
+                    rightChoice = true;
+                    break;
+                case "C":
+                    ShowEntryMenu();
+                    rightChoice = true;
+                    break;
+                case "X":
+                    rightChoice = true;
+                    break;
+                default:
+                    Console.WriteLine(@"Wrong choice. Try again please.");
+                    break;
+            }
+        }
+    }
```

Comment:
what I tried to do is to keep the user interaction inside the Menu class, this way it would work as the frontend of the application (I need to clean it further though). if it was a desktop application the switch would be buttons. which really emphasizes on your idea of storing the choices, so I can loop through them and display them accordingly.

Reply:
You don't necessarily need to store those elements in separate classes, but I would at least separate the displaying of the menu and the processing of the choice into two methods.